File: array/178.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> distributeCandies(int candies, int num_people) {
    vector<int> people(num_people, 0);
    int iteration = 0;

    while(candies > 0) {

        for(int i = 0; i < num_people; ++i) {
            if (candies <= 0)
                break;

            people[i] += min((num_people) * iteration + i + 1, candies);
            candies -= (num_people) * iteration + i + 1;

        }

        ++iteration;
    }

    return people;
}

int main() {
    int candies = 10, num_people = 3;
    for(const auto &elem: distributeCandies(candies, num_people))
        cout<<elem<<" ";

    return 0;
}

File: array/178.py
from typing import List


def distributeCandies(candies: int, num_people: int) -> List[int]:
    people = [0] * num_people
    iteration = 0

    while candies > 0:

        for index in range(num_people):
            if candies <= 0:
                break

            people[index] += min(num_people * iteration + index + 1, candies)
            candies -= num_people * iteration + index + 1

        iteration += 1

    return people


print(distributeCandies(10, 3))
